slatedb/src/admin.rs

```diff
@@ -275,6 +293,58 @@ impl Admin {
             .map_err(Into::into)
     }
 
+    /// Returns the timestamp or sequence from the latest manifest's sequence tracker.
+    /// When `round_up` is true, uses the next higher value; otherwise the previous one.
+    pub async fn get_timestamp_for_sequence(
+        &self,
+        seq: u64,
+        round_up: bool,
+    ) -> Result<Option<DateTime<Utc>>, crate::Error> {
+        let manifest_store = ManifestStore::new(
```

Comment:
nit: let's create a helper function to get the sequence tracker so we're not duplicating this code twice